maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/AbstractSurefireMojoToolchainsTest.java

```diff
@@ -0,0 +1,107 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.DefaultArtifact;
+import org.apache.maven.artifact.handler.ArtifactHandler;
+import org.apache.maven.artifact.versioning.VersionRange;
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.surefire.util.ReflectionUtils;
+import org.apache.maven.toolchain.ToolchainManager;
+import org.codehaus.plexus.languages.java.jpms.ResolvePathsRequest;
+import org.codehaus.plexus.logging.Logger;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.apache.maven.artifact.versioning.VersionRange.createFromVersion;
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+/**
+ * Test for {@link AbstractSurefireMojo}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({AbstractSurefireMojo.class, ResolvePathsRequest.class, ReflectionUtils.class})
+@PowerMockIgnore({"org.jacoco.agent.rt.*", "com.vladium.emma.rt.*"})
+public class AbstractSurefireMojoToolchainsTest
+{
+
+    private final AbstractSurefireMojo mojo = new AbstractSurefireMojoTest.Mojo();
+
+    @Before
+    public void setupMojo() throws Exception
+    {
+        MavenSession session = mock( MavenSession.class );
+        mojo.setSession( session );
+
+        MavenExecutionRequest mavenExecutionRequest = mock( MavenExecutionRequest.class );
+        when( session.getRequest() ).thenReturn( mavenExecutionRequest );
+
+        Logger logger = mock( Logger.class );
+        mojo.setLogger( logger );
+
+        final VersionRange surefireVersion = createFromVersion( "1" );
+        Artifact forkedBooter = new DefaultArtifact( "org.apache.maven.surefire",
+            "surefire-booter", surefireVersion, null, "jar", null, mock( ArtifactHandler.class ) );
+
+        mojo.setPluginArtifactMap( singletonMap( "org.apache.maven.surefire:surefire-booter", forkedBooter ) );
+    }
+
+    @Test
+    public void shouldHandleMaven30ToolchainManager() throws Exception
+
+    {
+        mockStatic( ReflectionUtils.class );
+        //TODO: mock this to return a method
```

Comment:
I think I was trying to test Maven 3.3 in this method... eventually.

Reply:
See my last [commit](https://github.com/apache/maven-surefire/commit/2fd2b330948e1d51e80d4830cbb00727ea3144ac) in my [branch](https://github.com/apache/maven-surefire/commits/pull/285). It's quite simple because the trick is in methods segregation and we can easily remove the method and tests in Maven 3.3.0.